infrastructure/tasks/10-functions.ts

```diff
@@ -67,69 +83,83 @@ export const run = async () => {
   );
   const servicePlan = await getAppServicePlan(webSiteClient);
 
-  await webSiteClient.webApps.createOrUpdate(
-    (config as any).azurerm_resource_group_00,
-    (config as any).azurerm_functionapp_00,
-    {
-      kind: "functionapp",
-      location: (config as any).location,
-      serverFarmId: servicePlan.id,
-      siteConfig: {
-        alwaysOn: false,
-        // nodeVersion:
-        // apiDefinition.url
-        appSettings: [
-          // mandatory params
-          { name: "AzureWebJobsStorage", value: storageConnectionString },
-          { name: "AzureWebJobsDashboard", value: storageConnectionString },
-          { name: "WEBSITE_NODE_DEFAULT_VERSION", value: "6.5.0" },
-          { name: "FUNCTIONS_EXTENSION_VERSION", value: "~1" },
-          // optional params
-          {
-            name: "COSMODB_NAME",
-            value: (config as any).azurerm_cosmosdb_documentdb_00
-          },
-          { name: "QueueStorageConnection", value: storageConnectionString },
-          { name: "APPINSIGHTS_INSTRUMENTATIONKEY", value: "" },
-          { name: "FUNCTION_APP_EDIT_MODE", value: "readonly" },
-          { name: "AzureWebJobsSecretStorageType", value: "disabled" },
-          { name: "WEBSITE_HTTPLOGGING_RETENTION_DAYS", value: "3" },
-          {
-            name: "MESSAGE_CONTAINER_NAME",
-            value: (config as any).message_blob_container_00
-          }
-        ],
-        connectionStrings: [
-          {
-            connectionString: cosmosdbLink,
-            name: "COSMOSDB_URI",
-            type: "Custom"
-          },
-          {
-            connectionString: cosmosdbKey,
-            name: "COSMOSDB_KEY",
-            type: "Custom"
-          }
-        ]
-      }
+  const appConfig = {
+    kind: "functionapp",
+    location: (config as any).location,
+    serverFarmId: servicePlan.id,
+    siteConfig: {
+      alwaysOn: true,
+      // nodeVersion:
+      // apiDefinition.url
+      appSettings: [
```

Comment:
why some settings are hardcoded here and some others are in `config` ?

Reply:
some settings should (must ?) have a fixed value (ie. it's a very bad idea to turn alwaysOn: false =)

Reply:
some others are fixed by default (ie "WEBSITE_NODE_DEFAULT_VERSION" *must* be "6.5.0")

Reply:
ok, you should add a comment here that explains what's the difference between these settings here and the ones in `config`, also explain why it's a bad idea to change these values (as somebody may think it's ok to change it)